arch/arm/src/common/Toolchain.defs

```diff
@@ -289,12 +289,18 @@ endif
 # Architecture flags
 
 ARCHCFLAGS += -Wstrict-prototypes -Wno-attributes -Wno-unknown-pragmas
-ARCHCXXFLAGS += -nostdinc++ -Wno-attributes -Wno-unknown-pragmas
+ARCHCXXFLAGS += -Wno-attributes -Wno-unknown-pragmas
 
 # When all C++ code is built using GCC 7.1 or a higher version,
 # we can safely disregard warnings of the type "parameter passing for X changed in GCC 7.1."
 # Refer to : https://stackoverflow.com/questions/48149323/what-does-the-gcc-warning-project-parameter-passing-for-x-changed-in-gcc-7-1-m
 
+ifneq ($(CONFIG_LIBCXXTOOLCHAIN),y)
+  ARCHCXXFLAGS += -nostdinc++
+else
+  ARCHCXXFLAGS += -D_STDLIB_H_
```

Comment:
why need `-D_STDLIB_H_`?

Reply:
This was the simplest work around for the error mentioned above (https://github.com/apache/nuttx/pull/11812#issuecomment-1973470015), the include guard prevents the multiple definition of the stdlib.h header file.